### PR TITLE
Drop support for MinGW (Issue #4593)

### PR DESCRIPTION
**Pull request by steven-johnson**
Not sure if we are ready for this yet or not, but here's a PR that removes all the MinGW if we decide to go that way. (Note that a separate PR for the buildbots will be necessary as well.)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,8 +133,7 @@ To configure and build Halide:
 
 #### Building Halide and LLVM on Windows using mingw
 
-The makefile method above should work from inside a "mingw64" shell
-(not the default shell) in an msys2 installation.
+Halide no longer supports building via the Makefile under mingw. Windows users should use the CMake build process instead.
```
Comment (abadams) [inline README.md]: This comment can be misinterpreted to mean that mingw is supported via cmake. We should just delete the mingw section.

File: Makefile [hunk 2]
```diff
@@ -964,21 +956,10 @@ all: distrib test_internal
...
-    MAP_FLAGS= -Wl,-t -Wl,-t
-    LIST_OUTPUT = 2>&1 | grep "libLLVM" | grep ")" > $(BUILD_DIR)/llvm_objects/list.all
 else
     MAP_FLAGS= -Wl,-Map=$(BUILD_DIR)/llvm_objects/list.all
     LIST_OUTPUT = > /dev/null
 endif
-endif
 
 $(BUILD_DIR)/llvm_objects/list: $(OBJECTS) $(INITIAL_MODULES)
 	# Determine the relevant object files from llvm with a dummy
```
Comment (shoaibkamil) [inline Makefile]: This could be more simplified if we don't support MinGW: We can get rid of `LIST_OUTPUT` and in line 985 always pipe to `/dev/null`.

**Comment by shoaibkamil** [comment]
I'm not sure if we should do this yet, but the PR does make a good case that removing MinGW simplifies several pieces of the build process.  Otherwise LGTM.

Perhaps we should send a message on the halide-dev list as you suggested?

File: Makefile (hunk 2 above)
Comment (steven-johnson) [reply Makefile]: Done. Also the removal of xargs later in the file.

**Comment by steven-johnson** [comment]
> Perhaps we should send a message on the halide-dev list as you suggested?

Done, a couple days ago. No responses as of yet.

**Comment by steven-johnson** [comment]
So far, not a single complaint about this proposal. How long should we wait for feedback?

**Comment by shoaibkamil** [comment]
I don't think we should wait too long, if nobody speaks up.  I'm ok with merging this, but maybe not on a Friday afternoon :)

**Comment by steven-johnson** [comment]
I just landed the change to the buildbots, so it's definitely not gonna get tested anymore. Might as well approve this one too :-)

**Comment by shoaibkamil** [comment]
Unfortunately LLVM looks like it's failing to build on the buildbots?

**Comment by shoaibkamil** [comment]
Oh DOH, that's because they were running when you landed the changes to the buildbot config.

**Pull request merged**